src/client/ua_client_connect.c

```diff
@@ -1513,6 +1538,141 @@ UA_Client_connectSecureChannel(UA_Client *client, const char *endpointUrl) {
     return client->connectStatus;
 }
 
+static void
+__Client_reverseConnectCallback(UA_ConnectionManager *cm, uintptr_t connectionId,
+                         void *application, void **connectionContext,
+                         UA_ConnectionState state, const UA_KeyValueMap *params,
+                         UA_ByteString msg) {
+    void *reverseConnectIndicator = (void *)(uintptr_t)0xFFFFFFFF;
+
+    UA_Client *client = (UA_Client*)application;
+
+    UA_LOCK(&client->clientMutex);
+
+    if (!client->channel.connectionId) {
+        client->channel.connectionId = connectionId;
+        *connectionContext = reverseConnectIndicator;
+    }
+
+    if (*connectionContext == reverseConnectIndicator && state == UA_CONNECTIONSTATE_CLOSING) {
```

Comment:
Add a small comment to avoid misunderstandings.

This is what I think is happening. Right?
So in the outer if the listen-socket is closing.
And in the inner if you check whether an active connection has been added since.
If there is an active connection (and we are shutting the listening socket down ourselves), do nothing.
Otherwise set the channel to "closed".

For the inner if `client->channel.connectionId == connectionId` should be enough.
As we immediately reset the connectionId when the active socket opens.

Reply:
Done